apps/gateway/lib/gateway/controllers/arena_servers_controller.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Gateway.Controllers.ArenaServersController do
+  @moduledoc """
+  Controller for health check.
+  """
+  use Gateway, :controller
+  alias GameBackend.Configuration
+
+  action_fallback Gateway.Controllers.FallbackController
+
+  def list_arena_servers(conn, _params) do
+    arena_servers =
+      Configuration.list_arena_servers()
+      |> Enum.map(fn arena_server ->
+        %{
+          id: arena_server.id,
+          ip: arena_server.ip,
+          url: arena_server.url,
+          name: arena_server.name,
+          status: arena_server.status,
+          environment: arena_server.environment
+        }
+      end)
```

Comment:
This isn't necessary if you add `@derive Jason.Encoder` right before defining the ArenaServer schema.
I'll add a comment where you should put it for more clarity.

Reply:
nice catch